### PR TITLE
Extend qanary pipeline as component

### PR DESCRIPTION
**Pull request by dschiese**
### Added

- new conditional Controller with root-mapping (moved from QanaryServiceController)
- property `pipeline.component`

### CODE REVIEW

**Comment by dschiese** [comment]
See https://github.com/dschiese/Qanary-quality-assurance for dedicated integration tests.

File: qanary_pipeline-template/pom.xml [hunk 1]
```diff
@@ -5,23 +5,25 @@
...
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
         <version>2.7.8</version>
     </parent>
     <properties>
         <java.version>11</java.version>
-        <spring-boot-admin.version>2.7.16</spring-boot-admin.version>
-        <qanary.version>[3.13.0,4.0.0)</qanary.version>
+        <spring-boot-admin.version>2.7.15</spring-boot-admin.version>
```
Comment (dschiese) [inline qanary_pipeline-template/pom.xml:17]: Change back

File: qanary_pipeline-template/pom.xml [hunk 2]
```diff
@@ -78,6 +80,17 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
```
Comment (dschiese) [inline qanary_pipeline-template/pom.xml:83]: Isn't used, can be deleted

File: qanary_pipeline-template/pom.xml [hunk 3]
```diff
@@ -142,26 +161,8 @@
...
-                <exclusion>
-                    <groupId>log4j</groupId>
-                    <artifactId>log4j</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.slf4j</groupId>
-                    <artifactId>slf4j-log4j12</artifactId>
-                </exclusion>
-            </exclusions>
+            <artifactId>jena-querybuilder</artifactId>
```
Comment (dschiese) [inline qanary_pipeline-template/pom.xml:164]: Isn't used either, can be deleted

**Comment by dschiese** [comment]
The maven tests fail due to a missing QanaryComponent implementation. However, I cannot figure out what's the cause. The pipeline shouldn't instantiate any QanaryComponent related class, hence, the missing dependencies error shouldn't occur, too. Unfortunately, locally everything works and the tests succeed.

1. Are the properties being respected? Normally, the QanaryComponent classes are only instantiated and added to the context, when the property `pipeline.is.component` is either true or non-existing. We pre-defined it as false.

File: qanary_pipeline-template/src/main/java/eu/wdaqua/qanary/web/QanaryComponentsManagementController.java [hunk 4]
```diff
@@ -39,103 +35,103 @@
...
+            "application/ld+json", "application/xml", "application/rdf+xml", "application/n-triples", "text/n3",
+            "text/turtle"})
+    @Operation(summary = "get the description of a registered Qanary component as RDF data (use header to define the format, default: JSON-LD)", //
+            operationId = "getServiceDescriptionOfComponent", //
+            description = "Returns set of RDF triples (presented in defined format), these triples describe the required inputs (data types) and expectedly produced outputs of the requested component. If the component is not available, then the request will result in an HTTP code 404 (not found).")
+    public ResponseEntity<String> getServiceDescriptionOfComponentAsJsonLD(HttpServletRequest request,
+                                                                           @PathVariable String componentName) {
+        String acceptHeader = request.getHeader("Accept");
+        logger.info("forward using the Accept header: {}", acceptHeader);
+        return getServiceDescriptionOfComponent(componentName, acceptHeader);
```
[the remark this answers is not in the document]
Comment (dschiese) [reply qanary_pipeline-template/src/main/java/eu/wdaqua/qanary/web/QanaryComponentsManagementController.java:82]: @anbo-de Did this occur before? As far as I can see, no reasonable changes were made in this file.
When changes are required: Done in this PR or create a new issue?

Besides that, the actions and tests should work now.

[the remark this answers is not in the document]
Comment (anbo-de) [reply qanary_pipeline-template/src/main/java/eu/wdaqua/qanary/web/QanaryComponentsManagementController.java:82]: @dschiese I think, the code fragment was checked due to the re-formatting and therefore the assumed change of the source. I think, you need to add `StringEscapeUtils.escapeHtml4()` or [`HtmlUtils.htmlEscape()`](https://docs.spring.io/spring-framework/docs/5.0.2.RELEASE/kdoc-api/spring-framework/org.springframework.web.util/-html-utils/index.html), cf. https://snyk.io/de/blog/preventing-xss-snyk-code/#:~:text=Mitigating%20XSS%20vulnerabilities%20with%20Snyk%20Code

**Comment by dschiese** [comment]
I believe the XSS problem should be solved now. All user-passed variables are now checked, or am I wrong?

**Pull request merged**